cmd/juju/application/storage.go

```diff
@@ -0,0 +1,311 @@
+// Copyright 2025 Canonical Ltd.
+// Licensed under the AGPLv3, see LICENCE file for details.
+
+package application
+
+import (
+	"fmt"
+	"io"
+	"sort"
+
+	"github.com/juju/cmd/v3"
+	"github.com/juju/errors"
+	"github.com/juju/gnuflag"
+	"github.com/juju/names/v5"
+
+	"github.com/juju/juju/api/client/application"
+	jujucmd "github.com/juju/juju/cmd"
+	"github.com/juju/juju/cmd/juju/config"
+	"github.com/juju/juju/cmd/modelcmd"
+	"github.com/juju/juju/cmd/output"
+	"github.com/juju/juju/storage"
+)
+
+const (
+	appStorageConfigSummary = "Displays or sets storage directives for an application."
+
+	appStorageConfigDoc = `
+To view all storage directives for the given application:
+
+    juju application-storage <application>
+
+	By default, the config will be printed in a tabular format. You can instead
+print it in ` + "`json`" + ` or ` + "`yaml`" + ` format using the ` + "`--format`" + ` flag:
+
+   	juju application-storage <application> --format json
+    juju application-storage <application> --format yaml
+
+To view the directive of a single storage name:
+
+    juju application-storage <application> <storage-name>
+
+To set storage directives on an application:
+
+    juju application-storage <application> <storagename1>=<storage-directive> <storagename2>=<storage-directive> ...
+
```

Comment:
If you link to the storage directive doc at the top as I suggest above, you can skip all the lines that try to define it again below. Basically, delete between this point and the examples.